dev/conductor/lib/repository.dart

```diff
@@ -473,16 +482,36 @@ class FrameworkRepository extends Repository {
         'cache',
       );
 
+  /// Tag [commit] and push the tag to the remote.
+  void tag(String commit, String tagName, String remote) {
+    assert(commit.isNotEmpty);
```

Comment:
asserts are fine if you expect them only for testing. But if these are real error cases you need to guard for from user provided input, they need to be exceptions instead. (don't rely on running with --enable-asserts unless you write main.dart to throw without them enabled)

Reply:
> asserts are fine if you expect them only for testing. But if these are real error cases you need to guard for from user provided input, they need to be exceptions instead. (don't rely on running with --enable-asserts unless you write main.dart to throw without them enabled)

I did write the entrypoint to throw without asserts enabled: https://github.com/flutter/flutter/blob/master/dev/conductor/bin/conductor.dart#L79

The tool is intended to be invoked via [a bash wrapper](https://github.com/flutter/flutter/blob/master/dev/conductor/bin/conductor), not only to ensure that asserts are enabled, but also so that the correct version of dart is used.

Reply:
SGTM